### DISCUSSION TITLE
FastAPI vs. Flask? Please vote

### DISCUSSION THREAD

**Discussion by drbacke** [Akkudoktor-EOS/EOS#119] [Polls]
FastAPI vs. Flask? 
In my opinion FastAPI ist more suitable for this project.

**Comment by drbacke** [comment]
We need a way to standardize and document the http/JSON API. 
Swagger might make the most sense for this, now the question is on what basis do we put all this?
Flask was my first option, as it was easy and familiar to me (as sole) developer. But that could become a problem in the long run.

**Reply by Lasall** [reply]
FastAPI comes with validation and provides Swagger ootb. By standardization you mean the API design?
As soon as statefulness becomes relevant (e.g. user authentication, DB storage) then Django/DRF might be the better choice.

**Comment by akokam** [comment]
I‘ve had good experience with FastAPI, pydantic and https for backend development! Only „downside“: this get async a bit and not everyone is used to it!

**Comment by eckelj** [comment]
FastAPI comes also with a nice open API that help you to communicate how to use it for those that are not too familiar with the domain
